File: src/main/java/com/naukma/cauliflower/entities/ServiceInstance.java
package com.naukma.cauliflower.entities;

import com.sun.xml.internal.ws.wsdl.writer.document.*;

import java.io.Serializable;
import java.util.GregorianCalendar;

/**
 * Created by Max on 27.11.2014.
 */
public class ServiceInstance implements Serializable {
    private int instanceId;
    private int userId;
    private ServiceLocation serviceLocation;
    private int serviceId;
 //   private Service service;
    private int instanceStatusId;
    private String instanceStatus;
    private int cableId;
    private boolean isBlocked;

    public ServiceInstance(int instanceId, int userId,
                           int serviceLocationId, String locationAddress,
                           int locationLongitude, int locationLatitude,
                           int serviceId, int instanceStatusId,
                           String instanceStatus, int cableId, boolean isBlocked) {
        this.instanceId = instanceId;
        this.userId = userId;
        serviceLocation = new ServiceLocation(serviceLocationId,locationAddress,locationLongitude,locationLatitude);
        this.serviceId = serviceId;
        this.instanceStatusId = instanceStatusId;
        this.instanceStatus = instanceStatus;
        this.cableId = cableId;
        this.isBlocked = isBlocked;
    //    this.service = null;
    }


    public ServiceInstance(int instanceId, int userId,
                           ServiceLocation serviceLocation, int serviceId,
                           int instanceStatusId, String instanceStatus,
                           int cableId, boolean isBlocked) {
        this.instanceId = instanceId;
        this.userId = userId;
        this.serviceLocation = serviceLocation;
        this.serviceId = serviceId;
        this.instanceStatusId = instanceStatusId;
        this.instanceStatus = instanceStatus;
        this.cableId = cableId;
        this.isBlocked = isBlocked;
    }

    public int getInstanceId() {
        return instanceId;
    }

    public int getUserId() {
        return userId;
    }

    public ServiceLocation getServiceLocation() {
        return serviceLocation;
    }

    public int getServiceId() {

        return serviceId;
    }


    public int getInstanceStatusId() {

        return instanceStatusId;
    }

    public String getInstanceStatus()
    {
        return instanceStatus;
    }

    public int getCableId() {

        return cableId;
    }
    public boolean getIsBlocked(){
        return this.isBlocked;
    }
}

File: webapp/web/js/ServiceInstance.js
function ServiceInstance(){
	var _id;
	var _userId;
	var _serviceLocation;
	var _serviceId;
	var _instanceStatusId;
	var _instanceStatus;
	var _cableId;
	var _isBlocked;
	var _arrServiceOrder = [];
	var _providerLocation;

	this.getId = function(){ return _id; }
	this.setId = function(id){ _id = id; }

	this.getUserId = function(){ return _userId; }
	this.setUserId = function(userId){ _userId = userId; }

	this.getServiceLocation = function(){ return _serviceLocation; }
	this.setServiceLocation = function(serviceLocation){ _serviceLocation = serviceLocation; }

	this.getServiceId = function(){ return _serviceId; }
	this.setServiceId = function(serviceId){ _serviceId = serviceId; }

	this.getInstanceStatusId = function(){ return _instanceStatusId; }
	this.setInstanceStatusId = function(instanceStatusId){ _instanceStatusId = instanceStatusId; }

	this.getInstanceStatus = function(){ return _instanceStatus; }
	this.setInstanceStatus = function(instanceStatus){ _instanceStatus = instanceStatus; }

	this.getCableId = function(){ return _cableId; }
	this.setCableId = function(cableId){ _cableId = cableId; }

	this.getIsBlocked = function(){ return _isBlocked; }
	this.setIsBlocked = function(isBlocked){ _isBlocked = isBlocked; }

	this.getArrServiceOrder = function(){ return _arrServiceOrder; }
	this.setArrServiceOrder = function(arrServiceOrder){ _arrServiceOrder = arrServiceOrder; }
	this.addServiceOrder = function(serviceOrder){ _arrServiceOrder.push(serviceOrder); }

	this.getProviderLocation = function(){ return _providerLocation; }
	this.setProviderLocation = function(providerLocation){ _providerLocation = providerLocation; }

	this.toJsonObj = function(){
		var arrServiceOrder = [];

		for(var i = 0; i < _arrServiceOrder.length; i++){
			arrServiceOrder.push(_arrServiceOrder[i].toJsonObj());
		}

		return {
			id: _id,
			userId: _userId,
			serviceLocation: _serviceLocation,
			serviceId: _serviceId,
			instanceStatusId: _instanceStatusId,
			instanceStatus: _instanceStatus,
			cableId: _cableId,
			isBlocked: _isBlocked,
			arrServiceOrder: arrServiceOrder,
			providerLocation: _providerLocation 
		}
	}
}